scripts/model_analysis.py

```diff
@@ -0,0 +1,887 @@
+
+    # If model_file_path is specified, collect all the tests in the model_file_path parent directory path
+    # and in the test_list will only include the tests matching with model_file_path,
+    # otherwise collect all the tests in the pytest_directory_path specified by the user
+    if model_file_path:
+        model_file_path_list = model_file_path.split("/")[:-1]
+        tests_directory_path = "/".join(model_file_path_list)
+    else:
+        tests_directory_path = pytest_directory_path
+
+    test_list = collect_all_pytests(tests_directory_path)
+
+    if model_file_path:
+        test_list = [test for test in test_list if test.split("::")[0] == model_file_path]
+
+    assert test_list != [], f"No tests found in the {tests_directory_path} path"
+
+    # Create a dictonary contains model_name as key and model tests(i.e include variant, task) as values
+    model_name_to_tests = {}
+    for test in test_list:
+        model_name = test.split("::")[0].split("/")[-1].replace(".py", "").replace("test_", "")
+        if model_name not in model_name_to_tests.keys():
+            model_name_to_tests[model_name] = [test]
+        else:
+            model_name_to_tests[model_name].append(test)
+
+    # Generate unique op test for the all collected test and save the models unique ops test information in the unique_ops_output_directory_path
+    model_output_dir_paths = []
+    for model_name, tests in model_name_to_tests.items():
+        model_output_dir_path = os.path.join(unique_ops_output_directory_path, model_name)
+        os.makedirs(model_output_dir_path, exist_ok=True)
+        model_output_dir_paths.append(model_output_dir_path)
+        for test in tests:
+            logger.info(f"Running the tests : {test}")
+            try:
+                result = subprocess.run(
+                    ["pytest", test, "-vss", "--generate-unique-op-tests"],
+                    capture_output=True,
+                    text=True,
+                    check=True,
+                    env=dict(
+                        os.environ,
+                        FORGE_EXPORT_TVM_GENERATED_UNIQUE_OP_TESTS_DETAILS="1",
+                        FORGE_EXPORT_TVM_GENERATED_UNIQUE_OP_TESTS_DETAILS_DIR_PATH=model_output_dir_path,
+                    ),
+                )
+                if result.returncode != 0:
+                    logger.error(f"Error while running the pytest:\n stdout: {result.stdout}\n stderr: {result.stderr}")
+                else:
+                    logger.info(f"Successfully generated and exported unique ops test")
+
+            except subprocess.CalledProcessError as e:
+                logger.error(f"Error while running the pytest:\n {e.output}")
+
+    return model_output_dir_paths
+
+
+def run_model_unique_op_tests_and_generate_markdowns(
+    model_output_dir_paths, markdown_directory_path, dump_failure_logs
+):
+    """
+    Execute unique operation tests for specified models, gather compiler support details and
```

Comment:
One of the future optimizations can be combine all unique tests, run them and populate model information. Running unique op tests per model will for sure add redundancy. However, let's have a separate issues for this optimization once functional logic is in place.

Reply:
Instead of running unique op test by model, we should collect the unique op tests from all the models and avoid running the duplicate unique op test, for example if we have add op with input shape (1, 32) and (1, 32) in resnet and yolo, we should run only once and populate the test details for both yolo and resnet.
Am I right? @nvukobratTT

Reply:
> Yup, that's correct :)) I'm certain there are many ops with sam input shape and configurations repeating. If we only run unique ones, we'll get an additional boost. However, this should be additional optimization.

Okay Nikola

Reply:
Created a separate issue for optimizing the model analysis script and generate nightly test - https://github.com/tenstorrent/tt-forge-fe/issues/761